File: solutions/swift/difference-of-squares/1/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
struct Squares {
    let num: Int
    init(_ num: Int) {
        self.num = num
    }

    var squareOfSum: Int {
        let square = (0 ... num).reduce(into: 0) { $0 += $1 }
        return square * square
    }

    var sumOfSquares: Int {
        return (0 ... num).map { $0 * $0 }.reduce(into: 0) { $0 += $1 }
    }

    var differenceOfSquares: Int {
        return squareOfSum - sumOfSquares
    }
}

File: solutions/go/difference-of-squares/1/difference_of_squares.go
// Package diffsquares contains methods for SquareOfSum and SumOfSquares calculation
package diffsquares

// SquareOfSum return square of the sum of the first natural numbers below number
func SquareOfSum(number int) int {
	result := 0
	for i := 0; i <= number; i++ {
		result += i
	}
	return result * result
}

// SumOfSquares return sum of the squares of the first natural numbers below number
func SumOfSquares(number int) int {
	result := 0
	for i := 0; i <= number; i++ {
		result += i * i
	}
	return result
}

// Difference return difference of the SquareOfSum and SumOfSquares results for the number
func Difference(number int) int {
	return SquareOfSum(number) - SumOfSquares(number)
}
